docker/checks.d/packageCanBeDownloaded.py

```diff
@@ -1,34 +1,93 @@
 #!/usr/bin/env python3
+# Require Python 3.8
+
+from urllib.request import urlopen
+from urllib.error import URLError, HTTPError
 
-import urllib2
 import xml.etree.ElementTree as ET
 
-try:
-    # first, try to import the base class from old versions of the Agent...
-    from checks import AgentCheck
-except ImportError:
-    # ...if the above failed, the check is running in Agent version 6 or later
-    from datadog_checks.checks import AgentCheck
+from datadog_checks.base.checks import AgentCheck
+
+
+def get_latest_version(versions):
+    '''
+        get_latest_version takes a list of Jenkins versions
+        then return the latest one
+    '''
+
+    results = []
+    for i in range(10):
```

Comment:
what is this doing?

Reply:
It split a Jenkins version into an array like "2.249.3" would become ["2","249", "3"] then we sort each component. I am looking for the latest version with the biggest component so 2.249.3 is bigger than 2.266 because 2.249.3 has three components.

In the current state, we have a maximum of 3 components but I set it to 10 to have a big value enough.

Reply:
can't you just iterate over the length of the array created by version.split?

Reply:
I don't know that length at that time, version availabe are coming from [maven-metadata.xml](https://repo.jenkins-ci.org/releases/org/jenkins-ci/main/jenkins-war/maven-metadata.xml).

which can be like "2.0", "2.176.1", "2.0-alpha-1", "1.509.2.JENKINS-14362-jzlib"

Reply:
I think I should limit to max 3 components

Reply:
I added comments to clarify what this function and I limited the number of components to 3